docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,527 @@
+
+* Main PR: https://github.com/napari/napari/pull/3378, superseded by
+  https://github.com/napari/napari/pull/3555
+* Other PRs:
+    * https://github.com/napari/napari/pull/3462
+    * https://github.com/napari/napari/pull/4185
+    * https://github.com/napari/napari/pull/4210
+    * https://github.com/napari/napari/pull/4221
+    * https://github.com/napari/napari/pull/4307
+    * https://github.com/napari/napari/pull/4309
+    * https://github.com/napari/napari/pull/4387
+    * https://github.com/napari/napari/pull/4444
+    * https://github.com/napari/napari/pull/4447
+
+
+Adding the missing pieces to `constructor` involves changes in four different projects:
+
+* `conda/constructor` itself, the tool that builds the installer
+* `conda/menuinst`, responsible of creating the shortcuts
+* `conda/conda`, which relies on `menuinst` to delegate the shortcut creation, but this code path
+  was only enabled on Windows so far
+* `conda-forge/conda-standalone-feedstock`, which freezes a `conda` install along with their
+  dependencies and adds a thin layer on top to communicate with the `constructor` installer during
+  the installation process.
+
+The full list of PRs is available on
+[this issue comment](https://github.com/napari/napari/issues/4248#issuecomment-1066574137).
+A complete list of features can be found in the packaging documentation [^napari-packaging-docs].
+
+### Milestone 3: Adding support for conda packages in the plugin manager
+
+Support for conda/mamba handling of plugin installations was implemented in a base
+PR and then extended with different PRs:
+
+* Initial PR: https://github.com/napari/napari/pull/2943
+* Other PRs that improved and extended the funcionality:
+    * https://github.com/napari/napari/pull/4074
+    * https://github.com/napari/napari/pull/3369
+    * https://github.com/napari/napari/pull/3288
+
+Adding the necessary functionality to the plugin dialog, including:
+
+* Adapting the current `Installer` class to support not only `pip` but also `conda/mamba` in a
+  single code base.
+* Adding support for `install`, `uninstall`, and `update` commands.
+* Populating the listing with data obtained from the napari hub API [^napari-hub-api].
+* Updating the listing to filter and mark plugins based on their availability on conda-forge.
+
+Some UX/UI improvements that are not part of this milestone but could be implemented in future
+releases include:
+
+* Making the plugin dialog a side panel / side dockwidget (similar to VS Code
+  [^vscode-extensions-ui])
+* Allow for simultaneous install of plugins. Currently multiple plugins can be selected for
+  install, uninstall and update, but each on of these actions, are queued and run sequentially.
+* Allow advanced users to install packages from PyPI or other local sources via drag-and-drop. This
+  presents many risks that could irreversibly disrupt the installation, though, so it needs to be
+  considered very carefully.
```

Comment:
This is a use case that I think we really want to support at some point, so the idea of "local conda packages" would be really nice to flesh out (happy for that to be a totally different doc/PR). Many labs will want to send plugins back and forth over email without publishing them anywhere.

Reply:
Yeah, for me, this has been and continues to be a _critical_ feature.  Until we have a way to support installs of plugins that are not publicly published on conda-forge or another public conda channel, I think we must support PyPI and/or wheels.

Reply:
> Many labs will want to send plugins back and forth over email without publishing them anywhere.

What about private Github repo? Or private conda channels?

Reply:
On conda world, the right way here is to use a custom channel (private or public). It can be a one-off setting (via command-line argument or env var), or permanent (we add it to napari's .condarc). This is technically possible right now already, but I guess some UI on top of those options would be nicer.

For dropped tarballs, ideally we create a temporary local channel on the spot, index it, and then install from there (+ deps from conda-forge).

Reply:
> For dropped tarballs, ideally we create a temporary local channel on the spot, index it, and then install from there (+ deps from conda-forge).

Love this.